### PR TITLE
People on fire will now emit fire colored light

### LINKED ISSUE #30475: People on fire don't have fire-colored light
They're just normal light.

### PR DESCRIPTION
**Pull request by ghost**
basically this is like #30490 except I blended the colors
capture

:cl: More Robust Than You
add: People on fire will now emit fire colored light
/:cl:

closes #30475

### CODE REVIEW

File: code/modules/mob/living/living.dm [hunk 1]
```diff
@@ -864,7 +864,10 @@
 		on_fire = 1
 		src.visible_message("<span class='warning'>[src] catches fire!</span>", \
 						"<span class='userdanger'>You're set on fire!</span>")
-		src.set_light(3)
+		var/lightcolor = LIGHT_COLOR_FIRE
+		if(light_power && light_range && light_color) //make sure the light is actually active!
+			lightcolor = BlendRGB(LIGHT_COLOR_FIRE, light_color, 0.6)
+		set_light(3, l_color = lightcolor)
```
Comment (ChangelingRain) [inline code/modules/mob/living/living.dm]: no, you can't just set the mob's light.

**Comment by ExcessiveUseOfCobblestone** [comment]
MORE LIGHT LAG FOR THE LAG GOD

File: code/modules/mob/living/living.dm (hunk 1 above)
Comment (MoreRobustThanYou) [reply code/modules/mob/living/living.dm]: What the hell would I do then?

Comment (ChangelingRain) [reply code/modules/mob/living/living.dm]: Make a fake object and put it in the mob.

Comment (MoreRobustThanYou) [reply code/modules/mob/living/living.dm]: that's retarded but ok

Comment (Cyberboss) [inline code/modules/mob/living/living.dm]: Fake object

**Review by Cyberboss — CHANGES_REQUESTED** [review]

**Pull request closed without merging**